core/templates/dev/head/collection_player/CollectionDataService.js

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to retrieve information about collections from the
+ * backend.
+ *
+ * @author henning.benmax@gmail.com (Ben Henning)
+ */
+
+// TODO(bhenning): For preview mode, this service should be omitted. Instead,
+// a separate collectionDataService should be created which serves a local copy
+// of the collection instead.
+oppia.factory('CollectionDataService', [
+    '$http', 'COLLECTION_DATA_URL', 'UrlInterpolationService',
+    function($http, COLLECTION_DATA_URL, urlInterpolationService) {
+
+  // Maps previously loaded collections to their IDs.
+  var _collectionCache = [];
```

Comment:
Per our offline discussion, we have decided to keep underscores for private members (variables and functions) of objects within JS. If I understand our conclusion, then this is a correct use of underscores.